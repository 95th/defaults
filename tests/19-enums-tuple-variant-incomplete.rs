use defaults::Defaults;

#[derive(Defaults)]
#[def = "A"]
enum Foo {
    A(usize),
    B,
}

fn main() {}
