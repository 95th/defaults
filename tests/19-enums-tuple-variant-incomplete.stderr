error[E0308]: mismatched types
 --> tests/19-enums-tuple-variant-incomplete.rs:4:9
  |
4 | #[def = "A"]
  |         ^^^
  |         |
  |         expected `Foo`, found enum constructor
  |         expected `Foo` because of return type
5 | enum Foo {
6 |     A(usize),
  |     - `A` defines an enum variant constructor here, which should be called
  |
  = note:          expected enum `Foo`
          found enum constructor `fn(usize) -> Foo {Foo::A}`
help: use parentheses to construct this tuple variant
  |
4 | #[def = "A"(/* usize */)]
  |            +++++++++++++
